torch/csrc/jit/serialization/source_range_serialization.cpp

```diff
@@ -65,10 +91,36 @@ c10::IValue SourceRangeSerializer::serialize_source(
   }
   c10::intrusive_ptr<c10::ivalue::Tuple> serialized;
   if (s == nullptr) {
-    serialized = c10::ivalue::Tuple::create({"", "", 0});
+    serialized = c10::ivalue::Tuple::create({-1, -1, 0});
   } else {
+    auto text_iter = text_to_pos_.find(s->text());
+    int64_t text_pos = -1;
+    if (text_iter == text_to_pos_.end()) {
+      text_pos = static_cast<int64_t>(texts_.size());
+      IValue text = s->text();
+      text_to_pos_[s->text()] = text_pos;
+      texts_.push_back(text);
```

Comment:
Might be cheaper to emplace_back `s_text()` directly

Reply:
done.